### ISSUE TITLE
auto-refresh db

### ISSUE THREAD

**Issue by tab1tha** [SocialFinanceDigitalLabs/family-context-live#107]
Create a feature that receives a location of data files and updates the database at intervals or whenever there is a file change in the specified directory.

**Comment by franciscobmacedo** [comment]
I'm not sure I understand this, is this supposed to work in production?

**Comment by tab1tha** [comment]
Hi Francisco, we want the app to be able to "watch" a directory (e.g sharepoint folder) and update the database at intervals or whenever that sharepoint folder is updated.
This is so that if the LA keeps track of updated values in a shared location, we are not asking them to go through an extra step to upload it to the tool. We can track the shared location that they already use and then update the database automatically. 
plan:
- I'm using watchdog to track changes file changes https://pypi.org/project/watchdog/ and then linking it to run the data_intake command logic.

Is there a way to do this better?
Do tell me what you think.

**Comment by tab1tha** [comment]
A few more ToDos:

- [ ] Add authentication coupling so that `pandas.read_csv()` works with protected files (e.g files on sharepoint).
- [ ] Sanitize data before uploading to Database to make the get_or_create process more robust. So that if "John Doe" is in the database, `get_or_create` doesn't add "John doe" as a different name.
- [ ] Implement json comparison. Decision needs to be made on whether existing data should be totally replaced with incoming data or updated only. For example, if existing json contains columns A, B, C and incoming data contains columns A, B, D, should we assume that column C has been deleted (result: A, B, D) or column D has been added (result: A, B, C, D)?. If we chose to totally replace (first case), then we require user uploads to always be a full representation of the current state of the whole population. On the other hand, if we only update (second case) then the users can upload only the data that has changed and assume the tool assumes that the rest of the data remains the same. I'm leaning towards the second case @cyramic 

I'll push a few more commits, to implement these. You both can redirect this at any point. @franciscobmacedo

**Comment by cyramic** [comment]
I put some of my thoughts on the [PR here](https://github.com/SocialFinanceDigitalLabs/family-context-live/pull/111#issuecomment-1832217206) about this one.  I'm hesitant, like I said with linking this to any specific tech or platform (e.g. Heroku) as the current nature of the project is to be as platform independent as possible (One client is running this on AWS, and another is currently loading onto Azure).  As such, any way of doing this process needs to be configurable to match the platform it's running on.  

How we update the JSON parts of the data all depends on the needs from the client that we're still working out.  I'll let you know when we know more as I'm waiting on a response on it.

**Comment by tab1tha** [comment]
I see what you mean. 
I'll put this on hold until we know more about what the user needs/specifications are.
